sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_connection_manager_async.py

```diff
@@ -75,4 +76,7 @@ def __init__(self, **kwargs):
 
 
 def get_connection_manager(**kwargs):
+    connection_mode = kwargs.get("connection_mode", _ConnectionMode.SeparateConnection)
```

Comment:
The `**kwargs` passed in here comes from the constructor of the `EventHubClient`. How is the caller supposed to pass in a valid value here given that the type is internal?

Reply:
We get either users' input, for instance, http_proxy, or the default value of uamqp if it's not in users' input

Reply:
What does the user's code look like if they want to use websockets?

Reply:
```python
EventHubConsumerClient(..., transport_type = TransportType.WebSocket)
```

Reply:
Not blocking for preview5, solving

Reply:
Sorry. Meant to ask about connection_mode, not transport.

Reply:
(not blocking the release, but I would still like to get an answer :))

Reply:
connection_mode is not for external users. So our documentation doesn't mention it. We use it.

Reply:
...so we smuggle a `**kwarg` between internal parts of our code? The signature of `get_connection_manager` is suspect to say the least. Why not take a single `connection_mode` optional positional parameter with a default value? As it is now, the code is not very maintainable.